src/TAMSEquationSystem.C

```diff
@@ -0,0 +1,615 @@
+/*------------------------------------------------------------------------*/
+/*  Copyright 2014 Sandia Corporation.                                    */
+/*  This software is released under the license detailed                  */
+/*  in the file, LICENSE, which is located in the top-level Nalu          */
+/*  directory structure                                                   */
+/*------------------------------------------------------------------------*/
+
+#include <AlgorithmDriver.h>
+#include <ComputeTAMSAvgMdotEdgeAlgorithm.h>
+#include <ComputeTAMSAvgMdotElemAlgorithm.h>
+#include <ComputeMetricTensorNodeAlgorithm.h>
+#include <ComputeSSTTAMSAveragesNodeAlgorithm.h>
+#include <EquationSystem.h>
+#include <EquationSystems.h>
+#include <Enums.h>
+#include <FieldFunctions.h>
+#include <NaluEnv.h>
+#include <NaluParsing.h>
+#include <Realm.h>
+#include <Realms.h>
+#include <Simulation.h>
+#include <SolutionOptions.h>
+#include <TAMSEquationSystem.h>
+#include <TimeIntegrator.h>
+#include <TurbViscSSTTAMSAlgorithm.h>
+
+#include <SolverAlgorithmDriver.h>
+
+// template for supp algs
+#include <AlgTraits.h>
+#include <kernel/KernelBuilder.h>
+#include <kernel/KernelBuilderLog.h>
+
+// stk_util
+#include <stk_util/parallel/Parallel.hpp>
+
+// stk_mesh/base/fem
+#include <stk_mesh/base/BulkData.hpp>
+#include <stk_mesh/base/Field.hpp>
+//#include <stk_mesh/base/FieldParallel.hpp>
+//#include <stk_mesh/base/GetBuckets.hpp>
+//#include <stk_mesh/base/GetEntities.hpp>
+//#include <stk_mesh/base/CoordinateSystems.hpp>
+#include <stk_mesh/base/MetaData.hpp>
+
+// stk_io
+#include <stk_io/IossBridge.hpp>
+
+// stk_topo
+#include <stk_topology/topology.hpp>
+
+// stk_util
+#include <stk_util/parallel/ParallelReduce.hpp>
+
+namespace sierra {
+namespace nalu {
+
+//==========================================================================
+// Class Definition
+//==========================================================================
+// TAMSEquationSystem - manages alpha pde system
+//==========================================================================
+//--------------------------------------------------------------------------
+//-------- constructor -----------------------------------------------------
+//--------------------------------------------------------------------------
+TAMSEquationSystem::TAMSEquationSystem(EquationSystems& eqSystems)
+  : EquationSystem(eqSystems, "TAMSEQS", "time_averaged_model_split"),
+    managePNG_(realm_.get_consistent_mass_matrix_png("adaptivity_parameter")),
+    avgVelocity_(NULL),
+    avgDensity_(NULL),
+    avgTkeResolved_(NULL),
+    avgDudx_(NULL),
+    metric_(NULL),
+    alpha_(NULL),
+    resAdequacy_(NULL),
+    avgResAdequacy_(NULL),
+    avgProduction_(NULL),
+    avgTime_(NULL),
+    avgMdotScs_(NULL),
+    avgMdot_(NULL),
+    metricTensorAlgDriver_(new AlgorithmDriver(realm_)),
```

Comment:
It's not super important, but switching to an automatically managed pointer like unique_ptr would be nice.

Reply:
good idea. done.

Reply:
Given that it is initialized in a class constructor list, why not just a plain member instead of a pointer? Then we won't even need `std::unique_ptr` :)

Reply:
righto, turned them into plain members.